consumer/consumer.go

```diff
@@ -81,7 +81,7 @@ func (c *Consumer) setup() {
 		}
 	}
 
-	if c.retryStrategy == nil {
+	if c.config != nil && c.retryStrategy == nil {
```

Comment:
Are you sure this shouldn't be the following?:
```suggestion
	if c.config == nil || c.retryStrategy == nil {
```

Reply:
No it shouldn't. 

What we want here is to create the `retryStrategy` if it's not already done.

So we need the `MaxRetryInterval` which is coming from the config but we should assert that the it is not nil to avoid panicking.

Reply:
Yes, I see that.

My concern really is this: does the lack of config imply that we shouldn't create a retryStrategy?  Is this valid?

Reply:
The thing is from where should we get the `MaxDelay` value if we don't have a config?

Reply:
@tealeg 🆙 ?